File: Python/253. Meeting Rooms II.py
class Solution:
    def minMeetingRooms(self, intervals: List[List[int]]) -> int:
        intervals = sorted(intervals)
        print(intervals)
        endQueue = list()
        for interval in intervals:
            if len(endQueue) != 0 and interval[0] >= endQueue[0]:
                heapq.heappop(endQueue)
            heapq.heappush(endQueue, interval[1])
        return len(endQueue)

File: 2021/253. Meeting Rooms II.cpp
class Solution {
public:
    int minMeetingRooms(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), [](auto& a, auto& b) {return a[0] < b[0]; });
        priority_queue<int, vector<int>, greater<int>> minHeap;
        for (vector<int>& interval : intervals) {
            if (minHeap.empty() || interval[0] < minHeap.top()) {
                minHeap.push(interval[1]);
            } else {
                minHeap.pop();
                minHeap.push(interval[1]);
            }
        }
        return minHeap.size();
    }
};